protocol/urls/urls.go

```diff
@@ -87,9 +97,23 @@ func GetLinkPreviewData(link string) (previewData LinkPreviewData, err error) {
 		return previewData, fmt.Errorf("Cant't parse link %s", link)
 	}
 
-	switch url.Hostname() {
-	case "youtube.com", "www.youtube.com", "youtu.be":
-		return GetYoutubePreviewData(link)
+	hostname := strings.ToLower(url.Hostname())
+	youtubeHostnames := []string{"youtube.com", "www.youtube.com", "youtu.be"}
+	for _, youtubeHostname := range youtubeHostnames {
+		if youtubeHostname == hostname {
+			return GetYoutubePreviewData(link)
+		}
+	}
+	for _, site := range LinkPreviewWhitelist() {
```

Comment:
I would probably consider `Tenor` and `giphy` special cases, as they are image websites, so I would be more explicit in handling them.
Either I would use the case statement we had before:
```
case "youtube.com", "www.youtube.com"....
case ""
```
But I take your concern is that you want to have subdomains as well (otherwise case statement should be fine).

If that's the case, maybe we can do something like:
```
if strings.HasSuffix(hostname, site.Address) && site.ImageWebsite {
   return GetImageWebsitePreviewData(link) 
}
```
and you'd have to add `ImageWebsite` bool to the struct

So that if someone adds a new website, that is not an image website, the function still throws an error, as we haven't implemented pulling `title`/`description` etc

What do you think?
Otherwise looks good to me, thank you!

Reply:
@cammellos, updated as per your suggestion.